acond/README.md

```diff
@@ -1,60 +1,192 @@
+
+`acond` can be built as either a static standalone or a dynamically linked executable.
+
+### Building `acond` in `rust` Container
+
+It's recommended to use the [`rust`][rust-docker] docker container image to build `acond` because it's easy to setup and also necessary to reproduce the `acond` binary.
+
+#### Prerequisites
+
+- [Bash](https://www.gnu.org/software/bash/) version 5 or higher - This is a standard component  in most Linux distros today.
+- [Docker Desktop](https://docs.docker.com/desktop/) - See [Install Docker Desktop on Linux](https://docs.docker.com/desktop/install/linux-install/) for detailed instructions.
+
+#### musl-libc Based Builds
+
+[musl-libc][musl-libc] is known for its small size. The container image `rust:alpine` is ideal for building [musl-libc](https://musl.libc.org) based executables.
+
+[`acon-build.env`](../scripts/acon-build.env) provides a collection of the bash functions to assist in building `acond` and initrd images. `start_rust_buildenv` is one of those shell functions and can be used to create an [Alpine][alpine-linux] based [`rust`][rust-docker] container, like below.
+
+```sh
+cd /path/to/ACON_PROJECT_ROOT
+. scripts/acon-build.env
+# NOTE: The optional U=. causes the container to setuidgid to current user's UID/GID
+U=. start_rust_buildenv
+```
+
+A command prompt (of the newly created `rust` docker container) will then come up and resemble the following.
+
+```
+ACON Repo       /path/to/ACON
+Rust OCI Image  rust:alpine
+INFO	acon-rust.USERNAME.0123456789ab: Creating new container...
+INFO	Installing downloaded packages in /acon/scripts/deps/rust-1.72.0/alpine ...
+INFO	To refresh dependent packages, simply delete '/acon/scripts/deps/rust-1.72.0/alpine/PACKAGES-INFO'
+fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
+fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz
+(1/11) Installing daemontools-encore (1.11-r1)
+(2/11) Installing libprotobuf (3.21.12-r2)
+(3/11) Installing libprotobuf-lite (3.21.12-r2)
+(4/11) Installing libprotoc (3.21.12-r2)
+(5/11) Installing musl-dev (1.2.4-r1)
+(6/11) Installing pkgconf (1.9.5-r0)
+(7/11) Installing openssl-dev (3.1.2-r0)
+(8/11) Installing openssl-libs-static (3.1.2-r0)
+(9/11) Installing zlib-dev (1.2.13-r1)
+(10/11) Installing protoc (3.21.12-r2)
+(11/11) Installing protobuf-dev (3.21.12-r2)
+Executing busybox-1.36.1-r2.trigger
+OK: 201 MiB in 38 packages
+/acon/acond $
+```
+
+Please note `start_rust_buildenv` identifies (using `git rev-parse --show-toplevel`) and maps `path/to/ACON_PROJECT_ROOT` to `/acon` inside the container, and sets the working directory to `/acon/acond` automatically. Therefore, building `acond` will be as simple as below.
+
+```sh
+./build
+```
```

Comment:
Should '$' be added before the command line? like -

$ ./build

Reply:
I'm not sure if it's a widely adopted convention, but I put in the prompt string only if what follows is the output of some program. In this case it's typed in by the user so I don't put in `$` here.